subsys/bluetooth/controller/ticker/ticker.c

```diff
@@ -492,6 +433,87 @@ static u32_t ticker_dequeue(struct ticker_instance *instance, u8_t id)
 	return (total + timeout);
 }
 
+/**
+ * @brief Resolve ticker node collision
+ *
+ * @details Evaluates the provided ticker node against other queued nodes
+ * and returns non-zero if the ticker node collides and should be skipped.
+ * The following checks are made:
+ *   1) If slot is reserved by previous node, skip is unconditional
+ *   2) If the periodic latency is not yet exhausted, node is skipped
+ *   3) If the node will starve next node due to slot reservation
+ *      overlap, node is skipped if:
+ *      a) Next node has higher priority than current node
+ *      b) Next node has more accumulated latency than the current node
+ *      c) Next node has force flag set, and the current does not
+ *
+ * @param nodes         Pointer to ticker node array
+ * @param ticker        Pointer to ticker to resolve
+ * @param slot_reserved Non-zero if slot is already reserved
+ *
+ * @return 0 if no collision was detected. 1 if ticker node collides
+ * with other ticker node of higher composite priority
+ * @internal
+ */
+static u8_t ticker_resolve_collision(struct ticker_node *nodes,
+				     struct ticker_node *ticker,
+				     u8_t slot_reserved)
+{
+	u8_t skipped = 0;
+
+	/* Check if node is being scheduled in reserved slot */
+	if (slot_reserved) {
+		skipped = 1;
+
+	} else if (ticker->lazy_periodic > ticker->lazy_current) {
+		skipped = 1;
+
+	} else if (ticker->next != TICKER_NULL) {
+		/* Check if this ticker node will starve next node which has
+		 * latency or higher priority
+		 */
+		s32_t lazy_current = ticker->lazy_current;
+
+		if (ticker->lazy_current >= ticker->lazy_periodic) {
+			lazy_current -= ticker->lazy_periodic;
+		}
+		u8_t id_head = ticker->next;
+
+		while (id_head != TICKER_NULL) {
+			struct ticker_node *ticker_next = &nodes[id_head];
+
+			s32_t lazy_next = ticker_next->lazy_current;
+
+			if (ticker_next->lazy_current >=
+			    ticker_next->lazy_periodic) {
+				lazy_next -= ticker_next->lazy_periodic;
+			}
+
+			u8_t force = (ticker_next->force > ticker->force);
+			u8_t has_priority =
```

Comment:
Here is effectively our priority policy/metric. But I am not sure adding the latency bias is The Best For All. It would be good to keep it simple with just priority, and outsource the policy to App (by having it change the priority).